app/models/concerns/orchestration/tftp.rb

```diff
@@ -87,7 +87,14 @@ def setTFTP(kind)
       logger.info "Deploying TFTP #{kind} configuration for #{host.name}"
       each_unique_feasible_tftp_proxy do |proxy|
         mac_addresses_for_provisioning.each do |mac_addr|
-          proxy.set(kind, mac_addr, :pxeconfig => content)
+          proxy.set(kind, mac_addr, {
+                      :pxeconfig => content,
+                      :targetos => host.operatingsystem.name.downcase,
+                      :major => host.operatingsystem.major,
+                      :minor => host.operatingsystem.minor,
+                      :arch => host.arch.bootfilename_efi,
```

Comment:
Let's use `host.arch.name` here which is at the end the architecture string reported by the system with `uname -m` (e.g. `x86_64`).

Reply:
Agreed, it's better to use `host.arch.name` here. => Changed and tested locally.